src/pages/RepoPage/SettingsTab/tabs/ConfigurationManager/hooks/useRepoConfigurationStatus/useRepoConfigurationStatus.tsx

```diff
@@ -0,0 +1,142 @@
+})
+
+const PlanSchema = z
+  .object({
+    tierName: z.nativeEnum(TierNames),
+  })
+  .nullable()
+
+export type RepositoryConfiguration =
+  | {
+      plan?: z.infer<typeof PlanSchema> | null
+      repository?: z.infer<typeof RepositorySchema>
+    }
+  | null
+  | undefined
+
+const RequestSchema = z.object({
+  owner: z
+    .object({
+      plan: PlanSchema,
+      repository: z.discriminatedUnion('__typename', [
+        RepositorySchema,
+        RepoNotFoundErrorSchema,
+        RepoOwnerNotActivatedErrorSchema,
+      ]),
+    })
+    .nullable(),
+})
+
+const query = `query GetRepoConfigurationStatus($owner: String!, $repo: String!){
+  owner(username: $owner) {
+    plan {
+      tierName
+    }
+    repository(name:$repo) {
+      __typename
+      ... on Repository {
+        flagsCount
+        componentsCount
+        coverageEnabled
+        bundleAnalysisEnabled
+        testAnalyticsEnabled
+        yaml
+      }
+      ... on NotFoundError {
+        message
+      }
+      ... on OwnerNotActivatedError {
+        message
+      }
+    }
+  }
+}`
+
+interface UseRepoConfigurationStatusArgs {
+  provider: string
+  owner: string
+  repo: string
+}
+
+export function useRepoConfigurationStatus({
+  provider,
+  owner,
+  repo,
+}: UseRepoConfigurationStatusArgs) {
+  return useQuery({
+    queryKey: ['GetRepoConfigurationStatus', provider, owner, repo],
+    queryFn: ({ signal }) => {
+      return Api.graphql({
+        provider,
+        signal,
+        query,
+        variables: {
+          owner,
+          repo,
+        },
+      }).then((res) => {
+        const parsedRes = RequestSchema.safeParse(res?.data)
+
+        if (!parsedRes.success) {
+          return Promise.reject({
+            status: 404,
+            data: {},
+            dev: 'useRepoConfigurationStatus - 404 Failed to parse data',
+          } satisfies NetworkErrorObject)
+        }
+
+        const data = parsedRes.data
+
+        if (data?.owner?.repository?.__typename === 'NotFoundError') {
+          return Promise.reject({
+            status: 404,
+            data: {},
+            dev: 'useRepoConfigurationStatus - 404 Not found error',
+          } satisfies NetworkErrorObject)
+        }
+
+        if (data?.owner?.repository?.__typename === 'OwnerNotActivatedError') {
+          return Promise.reject({
+            status: 403,
+            data: {
+              detail: (
+                <p>
+                  Activation is required to view this repo, please{' '}
+                  {/* @ts-expect-error */}
+                  <A to={{ pageName: 'membersTab' }}>click here </A> to activate
+                  your account.
+                </p>
+              ),
+            },
+            dev: 'useRepoConfigurationStatus - 403 Owner not activated error',
+          } satisfies NetworkErrorObject)
+        }
+
+        return {
+          plan: data?.owner?.plan,
+          repository: data?.owner?.repository,
```

Comment:
`l`: Could `?? null` these bad boys than you only have to handle `null` cases rather than both `null` and `undefined`

Reply:
great suggestion TY!